python-frontend/src/test/java/org/sonar/python/cfg/fixpoint/ReachingDefinitionsAnalysisTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.cfg.fixpoint;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.sonar.plugins.python.api.PythonFile;
+import org.sonar.plugins.python.api.tree.ExpressionStatement;
+import org.sonar.plugins.python.api.tree.FileInput;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.NumericLiteral;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.EXPRESSION_STMT;
+import static org.sonar.python.PythonTestUtils.getFirstDescendant;
+import static org.sonar.python.PythonTestUtils.getLastDescendant;
+import static org.sonar.python.PythonTestUtils.lastExpression;
+import static org.sonar.python.PythonTestUtils.lastExpressionInFunction;
+import static org.sonar.python.PythonTestUtils.parse;
+
+public class ReachingDefinitionsAnalysisTest {
+  private final PythonFile file = Mockito.mock(PythonFile.class, "file1.py");
+
+  @Test
+  public void valuesAtLocation_single_assignment() {
+    Name x = (Name) lastExpressionInFunction("x = 42; x");
+    ReachingDefinitionsAnalysis analysis = new ReachingDefinitionsAnalysis(file);
+    assertThat(analysis.valuesAtLocation(x)).extracting(expression -> ((NumericLiteral) expression).valueAsString()).containsExactly("42");
```

Comment:
What about factorizing this lambda? `expression -> ((NumericLiteral) expression).valueAsString()`